src/k8s/pkg/k8sd/api/cluster_bootstrap.go

```diff
@@ -0,0 +1,62 @@
+package api
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"time"
+
+	apiv1 "github.com/canonical/k8s/api/v1"
+	"github.com/canonical/k8s/pkg/utils"
+	"github.com/canonical/lxd/lxd/response"
+	"github.com/canonical/lxd/lxd/util"
+	"github.com/canonical/microcluster/microcluster"
+	"github.com/canonical/microcluster/state"
+)
+
+func postClusterBootstrap(m *microcluster.MicroCluster, s *state.State, r *http.Request) response.Response {
+	req := apiv1.ClusterBootstrapRequest{}
+	req.BootstrapConfig.SetDefaults()
+	if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
+		return response.BadRequest(fmt.Errorf("failed to parse request: %w", err))
+	}
+
+	//Convert Bootstrap config to map
+	config, err := req.BootstrapConfig.ToMap()
```

Comment:
You might want to wait for https://github.com/canonical/k8s-snap/pull/256 to land. I assume there might be some conflicts.